bundles/org.openhab.automation.scriptenginefactory.jython/src/main/java/org/openhab/automation/scriptenginefactory/jython/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.scriptenginefactory.jython;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.TreeSet;
+
+import javax.script.ScriptEngine;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.openhab.core.automation.module.script.AbstractScriptEngineFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * This is an implementation of {@link ScriptEngineFactory} for Jython.
+ *
+ * @author Scott Rushworth - Initial contribution
+ */
+@Component(service = org.openhab.core.automation.module.script.ScriptEngineFactory.class)
+@NonNullByDefault
+public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final String SCRIPT_TYPE = "py";
+    private static javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
+    private static final String DEFAULT_PYTHON_PATH = new StringBuilder(ConfigConstants.getConfigFolder())
+            .append(File.separator).append("automation").append(File.separator).append("lib").append(File.separator)
+            .append("python").toString();
```

Comment:
Wouldn't it be better to use the `java.nio.file.Paths` library to create this constant?

```suggestion
    private static final String DEFAULT_PYTHON_PATH = Paths.get(ConfigConstants.getConfigFolder() , "automation" , "lib" , "python").toString();
```

Reply:
I'm not sure if it is better, but it looks cleaner. I was modelling the code after ScriptFileWatcher. My concern is whether this will work properly on Windows, but I will test.